### ISSUE TITLE
Does not work on Archlinux since gnome-shell 3.34.0

### ISSUE THREAD

**Issue by felinae98** [phocean/TopIcons-plus#131]
When I enable it, I see the following in my journal:
```
9月 21 10:44:47 archlinux gnome-shell[1073]: JS ERROR: TypeError: ExtensionUtils.extensions is undefined
                                              onTrayIconAdded@/usr/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:84:1
9月 21 10:44:47 archlinux gnome-shell[1073]: JS ERROR: TypeError: ExtensionUtils.extensions is undefined
                                              onTrayIconAdded@/usr/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:84:1
9月 21 10:44:47 archlinux gnome-shell[1073]: JS ERROR: TypeError: ExtensionUtils.extensions is undefined
                                              onTrayIconAdded@/usr/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:84:1
```

**Comment by regymm** [comment]
Same here. 
```
Sep 21 12:09:50 localhost gnome-shell[1280]: Usage of object.actor is deprecated for ButtonBox
                                             get@resource:///org/gnome/shell/ui/environment.js:240:29
                                             createIconsContainer@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:171:5
                                             createTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:175:5
Sep 21 12:09:50 localhost gnome-shell[1280]: Usage of object.actor is deprecated for ButtonBox
                                             get@resource:///org/gnome/shell/ui/environment.js:240:29
                                             placeTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:280:9
                                             createTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:187:5
Sep 21 12:09:50 localhost gnome-shell[1280]: Usage of object.actor is deprecated for ButtonBox
                                             get@resource:///org/gnome/shell/ui/environment.js:240:29
                                             placeTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:282:9
                                             createTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:187:5
Sep 21 12:09:50 localhost gnome-shell[1280]: Usage of object.actor is deprecated for ButtonBox
                                             get@resource:///org/gnome/shell/ui/environment.js:240:29
                                             placeTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:294:5
                                             createTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:187:5
Sep 21 12:09:51 localhost dbus-daemon[516]: [system] Activating via systemd: service name='org.freedesktop.Avahi' unit='dbus-org.freedesktop.Avahi.service' r>
Sep 21 12:09:51 localhost dbus-daemon[516]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.Avahi.service': Unit dbus-org.freedesktop.A>
Sep 21 12:09:56 localhost systemd[1212]: gnome-terminal-server.service: Succeeded.
Sep 21 12:09:59 localhost gnome-shell[1280]: Usage of object.actor is deprecated for ButtonBox
                                             get@resource:///org/gnome/shell/ui/environment.js:240:29
                                             placeTray@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:280:9
                                             onPanelChange@/home/petergu/.local/share/gnome-shell/extensions/TopIcons@phocean.net/extension.js:135:5
                                             _onDestroy@resource:///org/gnome/shell/ui/panelMenu.js:90:9
                                             _onDestroy@resource:///org/gnome/shell/ui/panelMenu.js:189:9
                                             _onDestroy@/home/petergu/.local/share/gnome-shell/extensions/drive-menu@gnome-shell-extensions.gcampax.github.co>
                                             disable@/home/petergu/.local/share/gnome-shell/extensions/drive-menu@gnome-shell-extensions.gcampax.github.com/e>
                                             _callExtensionDisable@resource:///org/gnome/shell/ui/extensionSystem.js:78:13
                                             _onEnabledExtensionsChanged/<@resource:///org/gnome/shell/ui/extensionSystem.js:406:13
                                             _onEnabledExtensionsChanged@resource:///org/gnome/shell/ui/extensionSystem.js:403:9
                                             _onUserExtensionsEnabledChanged@resource:///org/gnome/shell/ui/extensionSystem.js:386:9
```

**Comment by davysson** [comment]
I had the same problem. Turns out I was using an old version of the extension, which broke after the update. So, to fix it just get the most recent version.

**Comment by regymm** [comment]
That works. Thanks. But I noticed another problem: The icons are shown only when I hover my cursor over it. Or the icons are not shown. But they can be used and right-clicked normally. 

I'm also using an extension called `appindicatorsupport`, it works without this problem, but some strange icons are not able to be shown(like icons of WINE applications), so if you don't use strange apps, this extension maybe a better choice.

**Comment by felinae98** [comment]
After upgrade from aur, it works. And I'm facing the same issue with ustcpetergu. I am searching for a tray addon which able to show the 'strange icons'.

**Comment by ElijahLynn** [comment]
> After upgrade from aur, it works. And I'm facing the same issue with ustcpetergu. I am searching for a tray addon which able to show the 'strange icons'.

How did you "upgrade from aur"?

**Comment by ElijahLynn** [comment]
I downloaded from source and ran the install manually (works, yay!) but now I am guessing you meant `yay -S gnome-shell-extension-topicons-plus` from https://aur.archlinux.org/packages/gnome-shell-extension-topicons-plus/.

**Comment by vhaidamaka** [comment]
It works from the master branch on gnome-shell 3.34.0-2 on Debian sid.

**Comment by gogogoghost** [comment]
I had the same problem and I solved it.
I met the same log: "ExtensionUtils.extensions is undefined"
See this link.
[stackoverflow](https://stackoverflow.com/questions/58085645/gnome-shell-3-34-missing-extensionutils-extension-property)
So I remove the code from line 83 to line 88 of extension.js, about the blacklist for skype. It works fine now.
I don't know how to  change it with new API, so I just remove it hahahaha. @felinae98

**Comment by joca-bt** [comment]
Maybe a new version can be released with the content of master and published on https://extensions.gnome.org?

**Comment by AsciiWolf** [comment]
> Maybe a new version can be released with the content of master and published on https://extensions.gnome.org?

@phocean Any news? The extensions.gnome.org version was still not updated.

**Comment by phocean** [comment]
This should have been fixed with #136 .
I will probably push a version on the Gnome website, but this is always a painful and slow process.

Feel free to reopen it if the issue still arises.
